pkg/service/id.go

```diff
@@ -61,7 +61,6 @@ func RestoreBackendID(l3n4Addr loadbalancer.L3n4Addr, id loadbalancer.BackendID)
 	// TODO(brb) This shouldn't happen (otherwise, there is a bug in the code).
 	//           But maybe it makes sense to delete all svc v2 in this case.
 	if newID != id {
-		DeleteBackendID(newID)
```

Comment:
Maybe you could add a comment saying that `newID` is used by some other restored backend which is used by a service? So releasing the ID would result in the interference.

Reply:
Good idea, I'll copy the text from the commit description.